Package lwt.5.7.0

raphael-proust opened the pull request:
### `lwt.5.7.0`
Promises and event-driven I/O
A promise is a value that may become determined in the future.

Lwt provides typed, composable promises. Promises that are resolved by I/O are
resolved by Lwt in parallel.

Meanwhile, OCaml code, including code creating and waiting on promises, runs in
a single thread by default. This reduces the need for locks or other
synchronization primitives. Code can be run in parallel on an opt-in basis.



---
* Homepage: https://github.com/ocsigen/lwt
* Source repo: git+https://github.com/ocsigen/lwt.git
* Bug tracker: https://github.com/ocsigen/lwt/issues

---
:camel: Pull-request generated by opam-publish v2.2.0

raphael-proust commented:
There are a few deprecated functions that are removed in this release so possibly a few packages that need additional constraints. I'll monitor the revdeps.

raphael-proust commented:
Notes for myself to prepare another PR adding constraints:

- `dune-rpc-lwt` uses `Lwt_result.catch`
- `crunch.2.2.0` uses `Result.result`
- `aws-s3-lwt` uses `Lwt_result.catch`
- `ocsigen-start` uses `Lwt_stream.Error`

Otherwise likely unrelated to this PR: `kinetic-client` seem to have some left-over bytes-vs-string build issues?

I think that's it. I'll open a PR for fixing the revdeps constraints.

yawaramin commented:
Out of curiosity, doesn't Lwt use Semantic Versioning?

raphael-proust commented:
It doesn't follow semantic versioning strictly. We do use the last digit for bug-fixes only but we're not too strict on the top two numbers.

sim642 commented:
On a somewhat related note, I'm wondering whether previous lwt versions should get an `"ocaml" {< "5.1"}` constraint due to https://github.com/ocsigen/lwt/pull/995.
Since [there are very few users of `Lwt_io.read_value`](https://sherlocode.com/?q=Lwt_io.read_value), it probably doesn't affect most Lwt users and could be overly restrictive to require Lwt >= 5.7 on OCaml 5.1. However, then downstream packages need to know about the incompatibility and add an opam conflict, in order to avoid crashes from marshaling.

raphael-proust commented:
> could be overly restrictive

We could add the restriction, and also ask potential users if they need a bugfix 5.6.2 release I guess.

I'm reluctant to do such a release if it's not used just because of time/resources, so I'll only do it if there is demand for it. Currently 5.1 is cutting edge so I don't know that it's unreasonable to expect users to be ok with a recent Lwt too.

raphael-proust commented:
I made https://github.com/ocaml/opam-repository/pull/24243 to mark incompatibilities. Note that pre-5.6 we already have the hard constraint for ocaml<5.0 so the bug affects only 5.6 so we might not need to do a bugfix release.

avsm commented:
Let's get this merged ahead of the 5.1 release. Is it just the upper bounds on use of Lwt_result.catch remaining, @raphael-proust ?

raphael-proust commented:
I think it's ready to be merged. The PR https://github.com/ocaml/opam-repository/pull/24228 addressed all the issues with `Lwt_result.catch` that I could find in the CI.

avsm commented:
In it goes!
